Rewrite heuristic to detect version

Nudin opened the pull request:
See #28

Nudin commented on main.py:
```diff
@@ -40,7 +42,7 @@
 
 class Settings:
-    do_update_wikidata = True
+    do_update_wikidata = False
```
This change was committed by accident. Will revert it. I think we should get these setting also out in the new settings-file.

konstin commented:
I've written a small tool to test the version handling with real data; I haven't looked thoroughly though it yet, but here are some remarks from the first inspection:
 
* I haven't found any false positives for stable versions, which is a great sign
* Sensitivity for stable versions also seems very good
* The `match = list(pre.finditer(string)) ` line is unsused
* Cases with `preview` as pseudo-semver classifier aren't handled
* Versions like `v1.0.0-beta3` with a leading `v` and a prerelease aren't recognised
* The unit tests are neat

Nudin commented:
I previously tested it with the main tool running in dry mode but this tool is indeed better. I added randomization and looked at several hundred cases. I found and fixed two cases were the string was wrongly sliced (including the case you mentioned). I after that couldn't find any cased where the recognized versions were wrong. There are some version-formats that are not yet detected but could be. But I think that is of low priority.

konstin commented:
I made a more thorough inspections and the only things I found are `9.3.1_RC2` and `2.7.3a0` which weren't detected as (pre)releases, but I agree that those are low priority.

All in all it looks awesome!

Nudin commented:
For the record these were the releases that I discovered that weren't detected:
    # release-1.6.4
    # release-2.1.8-stable
    # 2.0.180202
    # 2.7.3a0
    # v4.24-9651-beta

One still could consider trying to improve the handler. But merging for now.
